spec/support/helpers.rb

```diff
@@ -0,0 +1,142 @@
+require "tmpdir"
+require "open3"
+
+module Spec
+  module Helpers
+    def self.bundler_version
+      return @bundler_version if defined?(@bundler_version)
+      versions = Bundler.with_clean_env do
+        `gem list bundler | grep "bundler "`.chomp.scan(/\d+\.\d+\.\d+/)
```

Comment:
Just noticed this:  I guess if you have another `bundler...` gem, (like.... `bundler-inject`...), the grep would be needed.  Still can be done in Ruby though:

```ruby
`gem list bundler`.lines.grep("bundler ").chomp.scan(/\d+\.\d+\.\d+/)
```

Reply:
It's required for 2 reasons
a) `gem list bundler` has a `*** LOCAL GEMS ***` banner
b) if you have gems that have bundler in the name.  `gem list X` is a substring search so multiple gems show up.  Ironically, I have bundler-inject in my gem list, so when I do `gem list bundler` I get...

```text

*** LOCAL GEMS ***

bundler (2.0.1, 1.17.3, 1.16.6, 1.15.4)
bundler-inject (0.1.0)
```

Reply:
> would it be possible to do this without the shell out

I tried to figure that out and gave up for now...we can definitely do that in a follow up, but this works for now.

Reply:
> a) gem list bundler has a *** LOCAL GEMS *** banner

Huh, I originally tested this from an `irb` session, and it ended up not having that, but you are right that it shows up from a `bash` shell.  Weird.

Reply:
I don't see the difference, really, but yeah, I can change it to a Ruby grep.

Reply:
> I don't see the difference, really, but yeah, I can change it to a Ruby grep.

For portability, I tend to avoid pipes whenever possible, since it might not be supported, and it puts an external dependency on `grep` working how you would expect it to on your machine.

While not ideal, I am much more okay with a simple shell out to a command that we know is going to be there and perform roughly as we would expect (`gem` and `git` for example).  But any fancy processing I try to avoid doing in the shell when I am working in ruby (`bash` scripting is a different story, however).